.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -893,6 +899,17 @@ private DateTimeResolutionResult Match2Date(Match match, DateObject referenceDat
             ret.PastValue = futurePastDates.past;
             ret.Success = true;
 
+            // Ambiguous two-digit years are assigned values in both centuries (e.g. 35 -> 1935, 2035)
+            if (ambiguousCentury)
+            {
+                ret.PastValue = futurePastDates.past.AddYears(1900);
+                ret.FutureValue = futurePastDates.future.AddYears(2000);
+                if (ret.Timex.StartsWith("00"))
+                {
+                    ret.Timex = "XX" + ret.Timex.Substring(2);
```

Comment:
Timex creation should come from a method. Perhaps in TimexUtils. Pass the 2-year digit and the ambiguous flag and get the timex back.
Do we need this check for "00"? Won't all cases that fall here always be ambiguous? If we need the check, move it inside the method.

Reply:
moved Timex modification to TimexUtility.